awswrangler/_arrow.py

```diff
@@ -101,3 +101,11 @@ def _df_to_table(
                 table = table.set_column(col_index, field, table.column(col_name).cast(pyarrow_dtype))
                 _logger.debug("Casting column %s (%s) to %s (%s)", col_name, col_index, col_type, pyarrow_dtype)
     return table
+
+
+def _add_columns(table: pa.Table, schema: pa.Schema) -> pa.Table:
```

Comment:
This is really the main bandage here for the Modin issue. It does not currently handle the case when the columns are in a different order, or of the wrong types, but ideally I think we would like to add that at some point and make sure the table fully conforms to the unified schema.

Reply:
I'm not sure I understand the issue with columns being in different order? Do you mean in different order between the schemas?

Reply:
Yes, when different files have different column order.